seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/RowConverter.java

```diff
@@ -430,11 +429,7 @@ private OffsetDateTime convertOffsetDateTime(Object value) {
         } else if (value instanceof OffsetDateTime) {
             return (OffsetDateTime) value;
         } else if (value instanceof LocalDateTime) {
-            // Convert to OffsetDateTime using the system(jvm) default timezone
-            return ((LocalDateTime) value)
-                    .atZone(ZoneId.systemDefault())
-                    .withZoneSameInstant(ZoneOffset.UTC)
-                    .toOffsetDateTime();
+            return ((LocalDateTime) value).atOffset(ZoneOffset.UTC);
```

Comment:
This is not right, 
  // In Beijing (UTC+8), developer writes:
  LocalDateTime meeting = LocalDateTime.of(2024, 1, 1, 14, 0, 0); // 2 PM local

  // Original code: Correctly converts to 6 AM UTC
  // Modified code:  Treats as 2 PM UTC (8 hours off!)